### DISCUSSION TITLE
How does themes work on Windows?

### DISCUSSION THREAD

**Discussion by HubKing** [GtkSharp/GtkSharp#348] [Q&A] [answered]
I have tested the basic GtkSharp project. On Linux, of course, it followed the system theme. On Windows, it was the default light theme (Adwaita?) whether the system is in light or dark mode. I don't need to support a lot of themes, but I would like to have at least two themes (light and dark). How can this be achieved?

**Comment by hol430** [comment, marked as answer]
If themes are installed correctly, you should be able to set `GTK_THEME` environment variable to `Adwaita:dark` before starting your application.

https://docs.gtk.org/gtk3/running.html

**Reply by HubKing** [reply]
The thing that I would only need this when the OS is Windows (on Linux, it just works), and on Windows, I don't know how to install GTK themes nor can I expect users of my app to install GTK themes. So, I was wondering more like loading a theme from my app's directory and apply it to my app. If that is not a good way, how do I install a GTK theme on Windows?

**Comment by hol430** [comment]
I can't remember for sure, but I had thought that Adwaita:dark was supplied with the stock gtk installation on windows.

**Reply by HubKing** [reply]
There was no Adwaita:dark, but it seems that I can put theme files under `AppData\Local\Gtk\3.24.24\share\themes` and set it with Gtk.Settings.Default.ThemeName = "theme directory name";

**Comment by dsoronda** [comment]
If you use msys gtk instalation : find file `\msys64\mingw64\etc\gtk-3.0\settings.ini` (NOTE: Create file if doesn't exists !) and add line:

If you get GTK from GTK# , for Windows, location is `c:\Users\[your_user_name]\AppData\Local\Gtk\3.24.24\etc\gtk-3.0\settings.ini`

~~~ini
[Settings]
gtk-application-prefer-dark-theme=1
~~~

Note: 3.24.24 is GTK version installed so update accordingly.
